set.mm

```diff
@@ -751469,4 +751469,97 @@ with simplest antecedents (i.e., in the corresponding ordering of binary
 
 $( (End of Adhemar's mathbox.) $)
 
+$(
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+  Mathbox for Igor Ieskov
+#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#*#
+$)
+
+  ${
+    binom2d.1 $e |- ( ph -> A e. CC ) $.
+    binom2d.2 $e |- ( ph -> B e. CC ) $.
+    $( Deduction form of binom2.  (Contributed by Igor Ieskov, 14-Dec-2023.) $)
+    binom2d $p |- ( ph -> ( ( A + B ) ^ 2 ) =
+                ( ( ( A ^ 2 ) + ( 2 x. ( A x. B ) ) ) + ( B ^ 2 ) ) ) $=
+      ( cc wcel caddc co c2 cexp cmul wceq binom2 syl2anc ) ABFZGCPGBCHZIJZKZIB
+      RSIRBCLZITIQICRSIQIMDEBCNO $.
+  $}
+
+  ${
+    abc3i.1 $e |- A e. CC $.
+    abc3i.2 $e |- B e. CC $.
+    abc3i.3 $e |- C e. CC $.
+    $( Cube of sum of three numbers.  (Contributed by Igor Ieskov,
+       14-Dec-2023.) $)
+    abc3i $p |- ( ( ( A + B ) + C ) ^ 3 ) =
+                ( ( ( ( ( A ^ 3 ) + ( 3 x. ( ( A ^ 2 ) x. B ) ) )
+                + ( ( 3 x. ( A x. ( B ^ 2 ) ) ) + ( B ^ 3 ) ) )
+                + ( ( ( 3 x. ( ( A ^ 2 ) x. C ) )
+                + ( ( ( 3 x. 2 ) x. ( A x. B ) ) x. C ) )
+                + ( 3 x. ( ( B ^ 2 ) x. C ) ) ) )
+                + ( ( ( 3 x. ( A x. ( C ^ 2 ) ) )
+                + ( 3 x. ( B x. ( C ^ 2 ) ) ) ) + ( C ^ 3 ) ) ) $=
+      ( caddc wcel wceq addcli binom3 oveq1i eqtri oveq2i mulcli adddiri adddii
+      co sqcli mulassi c3 cexp c2 cmul cc wa pm3.2i ax-mp mp2an binom2i 2cn 3cn
+      eqcomi ) ABGZRZCUNRUAZUBZRZAUPUQRUPAUCZUQRZBUDZRVARUNRUPABUSUQRZVARVARBUP
+      UQRUNRUNRZUPUTCVARZVARZUPUSVARABVARZVARZCVARZUNRZUPVBCVARZVARZUNRZUNRZUPA
+      CUSUQRZVARZBVNVARZUNRZVARZCUPUQRZUNRZUNRZVMUPVOVARUPVPVARUNRZVSUNRZUNRURV
+      MUPUOVNVARZVARZVSUNRZUNRZWAURVCVEUPUSVFVARZCVARZVARZUNRZVKUNRZUNRZWFUNRZW
+      GURVCUPVDWIUNRZVARZVKUNRZUNRZWFUNRZWNURVCUPWOVJUNRZVARZUNRZWFUNRZWSURVCUP
+      UTWHUNRZCVARZVJUNRZVARZUNRZWFUNRZXCURVCUPXDVBUNRZCVARZVARZUNRZWFUNRZXIURV
+      CUPUOUSUQRZCVARZVARZUNRZWFUNRZXNURUOUPUQRZXQUNRZWFUNRZXSUOUEZHZCYCHZUFURY
+      BIYDYEABDEJFUGUOCKUHYAXRWFUNXTVCXQUNAYCHBYCHXTVCIDEABKUILLMXRXMWFUNXQXLVC
+      UNXPXKUPVAXOXJCVAABDEUJLNNLMXMXHWFUNXLXGVCUNXKXFUPVAXDVBCUTWHADSZUSVFUKZA
+      BDEOZOZJBESZFPNNLMXHXBWFUNXGXAVCUNXFWTUPVAXEWOVJUNUTWHCYFYIFPLNNLMXBWRWFU
+      NXAWQVCUNUPWOVJULZVDWIUTCYFFOZWHCYIFOZJVBCYJFOQNLMWRWMWFUNWQWLVCUNWPWKVKU
+      NUPVDWIYKYLYMQLNLMWMVMWFUNVMWMVLWLVCUNVIWKVKUNVHWJVEUNVHUPWHVARZCVARWJVGY
+      NCVAUPUSVFYKYGYHTLUPWHCYKYIFTMNLNUMLMWFVTVMUNWEVRVSUNWDVQUPVAABVNDECFSZPN
+      LNMVTWCVMUNVRWBVSUNUPVOVPYKAVNDYOOBVNEYOOQLNM $.
+  $}
+
+  ${
+    abc3d.1 $e |- ( ph -> A e. CC ) $.
+    abc3d.2 $e |- ( ph -> B e. CC ) $.
+    abc3d.3 $e |- ( ph -> C e. CC ) $.
+    $( Cube of sum of three numbers.  (Contributed by Igor Ieskov,
+       14-Dec-2023.) $)
+    abc3d $p |- ( ph -> ( ( ( A + B ) + C ) ^ 3 ) =
```

Comment:
Maybe I'm missing something, but doesn't it make more sense to prove the deduction version first and then derive `abc3i` as a corollary?

Reply:
I have changed the proof and reordered the theorems, so now it looks like the deduction was proved first and then the inference was derived from it. 

In reality, I proved the inference first, because it was easier for me. After that I created the proof for the deduction by "mechanically" wrapping each line of the inference proof into `ph -> ( ... )` and reproving everything. But reproving everything was not hard, because I had to remove all justifications and then the proof assistant found new ones. I only had to create binom2d and add a few new steps for this to work.